### ISSUE TITLE
Trying to build a model that is similar to opencv_ml's Boost implementation.

### ISSUE THREAD

**Issue by jknightmmcs** [mlpack/mlpack#2544]
Hello again, guys! 

I'm finally starting to write my own code with your library, and I'm stuck a bit.

My reason to move to mlpack was due to opencv_ml's implementation for most effective classifiers there (AdaBoost implementations - REAL, GENTLE, DISCRETE, and LOGIT) didn't provide probabilities for each class, and it became a huge problem for results analysis.

Now I have a problem:

I did manage to get mlpack's adaboost running with both Perceptron and DecisionStump as a weak learner, but actual classification accuracy isn't that great compared to what I had with cv::ml::Boost.

With all parameters in cv::ml::Boost(LOGIT or REAL) set as default, I had nearly 0.93 accuracy. With mlpack I've got those probabilities that I lacked with opencv, but accuracy has fallen down to 0.85 or so.

From what I can see, there's only a few parameters that I can tweak:
1) AdaBoost's iterations and tolerance
2) DecisionStump's bucket size
[ didn't dig too much into Perceptron, due to long learning time]

Tried to change them, didn't help that much.

Is there a way to build models similar to what cv::ml::Boost provides with this library?

**Comment by rcurtin** [comment]
Hey @jknightmmcs, sorry for the slow response on this one.

Can I ask what version of mlpack you are using?  At some point between mlpack 2 and 3 we shifted from using a custom decision stump implementation to a decision tree implementation that more closely reflects ID3.  If you can use a newer version (like 3.3.2 or git master), that should use the decision tree implementation.

One of the things about AdaBoost is that it can achieve perfect accuracy on the training set; when you say that accuracy is only 0.85, do you mean on a separate test set or on a training set?  If it's not achieving 1.0 accuracy on the training set, then I believe that probably the iterations or tolerance of AdaBoost need to be tuned.

**Comment by jknightmmcs** [comment]
I was using version 3.2.2. You suggest an update?

0.85 was achieved with a separate test set. I didn't actually check performance with a training set: as far as I know, measurements that only based on a training set are a best way for model to overfit.

Actually, I'm using cross-validation to check for model efficiency: dataset is splitted into 5 parts, one is for test, four are for training, repeated 5 times, using mean of success rates as a total model success rate.

**Comment by jknightmmcs** [comment]
Well, with DicisionTree weak learner I have finally got what I wanted - success rate is just about what I need, 0.93 on the same dataset. Thanks for the help :-)

**Comment by rcurtin** [comment]
@jknightmmcs great to hear you got it worked out! :+1:
